### PR TITLE
MIP45 post-audit updates

### PR DESCRIPTION
**Pull request by kmbarry1**
~Still a WIP; feedback/contributions welcome.~ Pretty much done, just waiting to be able to link to all audit reports.

### CODE REVIEW

**Comment by godsflaw** [comment]
Also needs:
[ ]  Add links to the IPFS documentation
[ ] `getId()` was removed update the documentation
[ ] `updust` must be added to the documentation
[ ] Some previously mentioned inconsistencies from PwC haven't been fixed yet
[ ] A mention of `ExponentialDecrease` contract
[ ] Possibly point to reference implementation of `clipperCallee` and the keeper from INT.

**Comment by kmbarry1** [comment]
Copying the list @godsflaw posted above so that I have one I can edit and add to:

[ ] Add links to audit reports.
[x] getId() was removed update the documentation
[x] updust must be added to the documentation
[x] Some previously mentioned inconsistencies from PwC haven't been fixed yet (6.3 in latest report)
[x] A mention of ExponentialDecrease contract
[x] Link to the reference implementation of clipperCallee and the keeper from INT.
[x] Mention PwC informational finding 8.2 ("Creation of Many Small Auctions") in section on incentive farming
[x] Breaker changes from dss#220
[x] Make sure the importance of `max` parameter is emphasized since prices can jump non-monotonically, e.g. on auction parameter changes (PwC 8.7).
[x] Make sure risk of oracle price delay allowing price to run away to the downside is adequately mentioned.
[x] Note that `tail` and `cusp` can change in-flight, if not already done.
[x] Make sure precision of all inputs/outputs/parameters is clearly specified.
[x] `updust` --> `upchost`

**Comment by godsflaw** [comment]
@gbalabasquer is suggesting we make the English description of `cut` more clear.

if `cut` is 1%, it means we take a 1% drop every `step` seconds.  Even though we specify the parameter as 0.99 WAD, we should make it easier to reason about for risk and governance.

File: MIP45/mip45.md
Comment (gbalabasquer) [inline MIP45/mip45.md]: This is `2`

File: MIP45/mip45.md
Comment (gbalabasquer) [inline MIP45/mip45.md]: This is `3`

File: MIP45/mip45.md
Comment (kmbarry1) [reply MIP45/mip45.md]: Good catch! Fixed.

File: MIP45/mip45.md
Comment (kmbarry1) [reply MIP45/mip45.md]: ditto

**Comment by kmbarry1** [comment]
This is at a point where it can be reviewed--the only remaining changes are trivial and blocked on actions outside my control (adding links to final audit reports and the example DEX bidding integration).

**Comment by Davidutro** [comment]
Looks good to me, format-wise.

@CPSTL would be good for you to look it over as well before we merge.

**Comment by CPSTL** [comment]
> This is at a point where it can be reviewed--the only remaining changes are trivial and blocked on actions outside my control (adding links to final audit reports and the example DEX bidding integration).

SGTM - I'm holding off on merging this PR until we get those final additions in plus a final sign-off from a member of the Smart Contract domain team.

File: MIP45/mip45.md
Comment (godsflaw) [inline MIP45/mip45.md]:
```suggestion
Parameters, e.g. `tail`, `cusp`, or the price decrease function or any of its parameters, can be changed at any time by governance and will affect the behavior of running auctions. Integrators should take this possibility into account, reasoning through how sudden change in parameters would impact their bidding strategies. Governance should endeavor to change parameters infrequently and if possible, only when there are not any auctions that will be affected.
```

Comment (godsflaw) [reply MIP45/mip45.md]: love this typo BTW.

**Review by godsflaw — APPROVED** [review]
LGTM

**Comment by kmbarry1** [comment]
Linked to all finalized audit reports we have received to date. If there's a strong desire to merge this, we can go ahead, and will add the link to the last audit report in a separate PR.

**Comment by CPSTL** [comment]
> Linked to all finalized audit reports we have received to date. If there's a strong desire to merge this, we can go ahead, and will add the link to the last audit report in a separate PR.

SGTM. Merging in updates 👍

**Pull request merged**
